imagescan/delta.go

```diff
@@ -293,6 +297,13 @@ func (d *deltaState) findBestNode(nodeNames []string, requiredMemory *inf.Dec, r
 	return candidates[0].name, nil
 }
 
+func (d *deltaState) nodeCount() int {
+	d.mu.RLock()
```

Comment:
What does this lock achieve? 🤔

Reply:
this allows to get number of nodes. we do determine image scan concurrency level based on that